pkg/archive.py

```diff
@@ -422,7 +425,14 @@ def close(self):
     # Close the gzip file object if necessary.
     if self.fileobj:
       self.fileobj.close()
-    if self.use_xz_tool:
+    if self.compressor:
```

Comment:
This approach wastes time and space because it fully writes the fie uncompress, then calls the compressor.
Instead, this should be done with a passed in fileobj the way we do for gz format.

You could wrap the compressor invocation as a subprocess.run(), where you forward writes to the fileobj to stdin of the process.

Reply:
Isn’t this the exact same approach used by the `xz` code below?

Reply:
This is resolved, and the `xz` code migrated to use the same (improved) code path.